File: 2022/56. Merge Intervals/Solution2.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals,  (x, y) -> Integer.compare(x[0], y[0]));
        List<int[]> list = new LinkedList<>();  
        for(int[] i : intervals) {
            if (list.isEmpty() || list.get(list.size() - 1)[1] < i[0])  {
                list.add(i);
            } else {
                list.get(list.size() - 1)[1] = Math.max(list.get(list.size() - 1)[1], i[1]);
            }
        }
        return list.toArray(new int[list.size()][2]);
    }
}

File: 2025/309. Best Time to Buy and Sell Stock with Cooldown/solution2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        dp = [[[0] * 2 for _ in range(2)] for _ in range(len(prices) + 1)]
        for i in range(len(prices) - 1, -1, -1):
                dp[i][1] = dp[i + 1][0]
                dp[i][0][0] = max(-prices[i] + dp[i + 1][0][1], dp[i + 1][0][0])
                dp[i][0][1] = max(prices[i] + dp[i + 1][1][0], dp[i + 1][0][1])
        return dp[0][0][0]